torchmetrics/metric.py

```diff
@@ -221,6 +234,77 @@ def wrapped_func(*args, **kwargs):
 
         return wrapped_func
 
+    def sync(
+        self,
+        dist_sync_fn: Optional[Callable] = None,
+        process_group: Optional[Any] = None,
+        should_sync: bool = True,
+        distributed_available: Optional[Callable] = distributed_available,
```

Comment:
n00b question: why does distributed available need to be an argument?

Reply:
Because `torchmetrics` does not know about any distributed platforms other than CUDA

For example TPU, IPUs...